fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/common/StoreTimer.java

```diff
@@ -167,6 +167,13 @@ protected static Counter getCounter(@Nonnull Map<Event, Counter> counters, @Nonn
          * @return the user-visible title
          */
         String title();
+
+        /**
+         * Get the logKey of this event for logging purposes.
+         *
+         * @return the key to use for logging
+         */
+        String logKey();
```

Comment:
It seems to me like this should default to the name or the name in all lower case or something reasonable like that. I think that would mean that implementors of the `Event` interface wouldn't need to change unless they wanted to use the new feature.

It also seems like some of the implementations can return `null`. If so, we should mark this as `@Nullable` explicitly. Alternatively, we may want to make it `@Nonnull` by contract and fix the implementations (which I think would all be fixed by either using the default implementation or by having constructors which set the log key).

Reply:
* Fixed the constructors and made the 'logKey()' methods @Nonnull. Please review.
* I did not add a call to '.toLower()' since the call sites seem to be doing the conversion to lower case. Thoughts?
* Added a test. Please review.

Reply:
I think maybe having `logKey` default to lower case is a good idea.  I know I sometimes harp on seemingly minor performance things, but `toLower()` is not a free call, and doing it at the call site means that we are doing a new string allocation, along with unicode case conversation logic (which I believe is not as simple as one would hope), on each and every call, where as we could be doing it once in the constructor.

Reply:
I think my preference would also be to have the `Event` store it in lower case rather than have the caller store it. That would be just to support something like `KeyValueLogMessage.of(staticMessage, event.logKey(), someEventBasedValue)` without the caller having to remember that the event log key should be minusculed to match other uses.

Reply:
Fixed. Please review.

Reply:
Okay left a comment about whether user-provided overrides should be lower-cased or not, but the default seems reasonable.

The other thing that maybe got lost is that I think the interface here should implement this with a `default` method that returns `name().toLowerCase(Locale.ROOT)` so that anyone who's implemented this interface doesn't have their code break when they upgrade. Obviously, that means that the value isn't memoized, but I think it gets them to the same place they are today.

Reply:
Done. Please review.